### ISSUE TITLE
GDScript 2.0: Specifing a typed array as return type of function results in error when type inside an array is a custom class

### ISSUE THREAD

**Issue by Trogloid** [godotengine/godot#59879]
### Godot version

4.0.dev

### System information

Windows 10

### Issue description

Specifing a typed array as return type of function throws an error when type inside an array is a custom class.
(works fine with built-ins)

![typedArray](https://user-images.githubusercontent.com/31877904/161585600-72f71374-c097-4a63-a4ef-661e1a447314.png)

### Steps to reproduce

```gdscript
extends Resource
class_name TextureData

func get_textures() -> Array[TextureData]:
    pass;
```

**Comment by cridenour** [comment]
I can confirm as of alpha14 (and possibly earlier) this has been fixed. Like closed as #63325 and can be closed.

**Comment by kleonc** [comment]
As pointed out by @cridenour it's already fixed by #63325 (seems like it's a duplicate of #59485 in the first place).
